src/rgw/rgw_rest_pubsub.cc

```diff
@@ -320,7 +308,8 @@ class RGWPSCreateTopicOp : public RGWOp {
       }
     } else {
       // if no topic policy exists, just check identity policies for denies
-      constexpr bool mandatory_policy = false;
+      // account users require an Allow, non-account users just check for Deny
+      const bool mandatory_policy = !!s->auth.identity->get_account();
```

Comment:
`!!` not of not ? or its typo ?

Reply:
brace-initialization also works:
```c++
const bool mandatory_policy{s->auth.identity->get_account()};
```

Reply:
or other option was to wrap the `verify_user_permission(` call only if its account as anyways we check user_permission if user_type == account ?